platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AbstractJavaCompileSpecFactory.java

```diff
@@ -81,7 +81,7 @@ private T chooseSpecFromCompileOptions(File fallbackJavaHome) {
         if (toolchain != null) {
             languageVersion = toolchain.getLanguageVersion().asInt();
         } else {
-            languageVersion = JavaVersion.current().ordinal() + 1;
+            languageVersion = JavaVersion.current().getMajorVersionNumber();
```

Comment:
I see we have a couple of places in the codebase where we do something like this:

    Integer.parseInt(JavaVersion.current().getMajorVersion())

You might want to replace those with `getMajorVersionNumber()` (hint: search for the `ParseInt.*version` regex)

Reply:
Ah, Justin made the same comment. Sorry for the noise :)

Reply:
Maybe I've missed those, because I was focused on finding the ordinal() + 1 use cases.
I will have a second look